### PR TITLE
Feat/arborist

### PR DESCRIPTION
**Pull request by paulineribeyre**
- Use Arborist + gen3authz for auth requests
  - Use arborist's auth/mapping endpoint to get all the resources the user has access to, and parse the result to get the programs and projects the user has "peregrine read" access to
  - Patch `run.py` so that gen3authz requests are mocked when we run peregrine locally
- Unit tests:
  - Bump sheepdog to 2.0.0
  - Remove `project_access` from the test user tokens
  - Remove the `random_user` fixture (only keep `submitter` and `admin`) since authorization doesn't depend on project access in the token anymore
  - Delete unused file `tests/auth_mock.py`

### Breaking Changes
- Use Arborist + gen3authz for auth requests

### Dependency updates
- Add gen3authz dependency for auth requests

### Deployment changes
- Optional configuration variable ARBORIST (arborist base URL)
- Requires Arborist to be deployed

### CODE REVIEW

**Comment by PlanXCyborg** [comment]
Jenkins Build 1 : time taken 32 min
Check the https://jenkins.planx-pla.net/job/Performance%20tests%20for%20uc-cdis/job/peregrine/job/PR-138/1/display/redirect


Test results:
	Passed: 109, Failed: 0, Skipped: 0



| Test | Time (`PR`) | Time (`master`) | Diff |
| - | - | - | - |
| @ExportPerformanceTests @Performance: Export all nodes of type: aliquot | 0.201 | 0.22 | -0.02 |
| @ExportPerformanceTests @Performance: Export all nodes of type: case | 0.263 | 0.365 | -0.10 |
| @ExportPerformanceTests @Performance: Export all nodes of type: experiment | 0.4 | 0.303 | 0.10 |
| @ExportPerformanceTests @Performance: Export all nodes of type: read_group | 0.37 | 0.447 | -0.08 |
| @ExportPerformanceTests @Performance: Export all nodes of type: read_group_qc | 0.084 | 0.103 | -0.02 |
| @ExportPerformanceTests @Performance: Export all nodes of type: sample | 0.504 | 0.351 | 0.15 |
| @ExportPerformanceTests @Performance: Exporting a record by ID on nodes of type: case | 0.332 | 0.303 | 0.03 |
| @ExportPerformanceTests @Performance: Exporting a record by ID on nodes of type: experiment | 0.238 | 0.32 | -0.08 |
| @ExportPerformanceTests @Performance: Exporting a record by ID on nodes of type: read_group | 0.124 | 0.075 | 0.05 |
| @ExportPerformanceTests @Performance: Exporting a record by ID on nodes of type: read_group_qc | 0.062 | 0.142 | -0.08 |
| @ExportPerformanceTests @Performance: Exporting a record by ID on nodes of type: sample | 0.091 | 0.325 | -0.23 |
| @QueryPerformanceTests @Performance: Executing bottomUp query # 0 | 0.669 | 0.493 | 0.18 |
| @QueryPerformanceTests @Performance: Executing bottomUp query # 1 | 0.597 | 0.539 | 0.06 |
| @QueryPerformanceTests @Performance: Executing bottomUp query # 2 | 0.645 | 0.434 | 0.21 |
| @QueryPerformanceTests @Performance: Executing bottomUp query # 3 | 0.591 | 0.369 | 0.22 |
| @QueryPerformanceTests @Performance: Executing bottomUp query # 4 | 0.391 | 0.303 | 0.09 |
| @QueryPerformanceTests @Performance: Executing bottomUp query # 5 | 0.347 | 0.2 | 0.15 |
| @QueryPerformanceTests @Performance: Executing bottomUp query # 6 | 0.209 | 0.132 | 0.08 |
| @QueryPerformanceTests @Performance: Executing bottomUp query # 7 | 0.109 | 0.074 | 0.03 |
| @QueryPerformanceTests @Performance: Executing topDown query # 0 | 0.089 | 0.095 | -0.01 |
| @QueryPerformanceTests @Performance: Executing topDown query # 1 | 0.203 | 0.137 | 0.07 |
| @QueryPerformanceTests @Performance: Executing topDown query # 2 | 0.18 | 0.095 | 0.09 |
| @QueryPerformanceTests @Performance: Executing topDown query # 3 | 0.196 | 0.245 | -0.05 |
| @QueryPerformanceTests @Performance: Executing topDown query # 4 | 0.417 | 0.224 | 0.19 |
| @QueryPerformanceTests @Performance: Executing topDown query # 5 | 0.381 | 0.279 | 0.10 |
| @QueryPerformanceTests @Performance: Executing topDown query # 6 | 0.564 | 0.376 | 0.19 |
| @QueryPerformanceTests @Performance: Executing topDown query # 7 | 0.635 | 0.448 | 0.19 |
| @SubmissionPerformanceTests @Performance: Submission acknowledgement 10 | 0.462 | 0.311 | 0.15 |
| @SubmissionPerformanceTests @Performance: Submission acknowledgement 100 | 2.872 | 2.03 | 0.84 |
| @SubmissionPerformanceTests @Performance: Submission acknowledgement 1000 | 30.253 | 19.593 | 10.66 |
| @SubmissionPerformanceTests @Performance: Submission aligned_reads_index 100 | 18.413 | 17.276 | 1.14 |
| @SubmissionPerformanceTests @Performance: Submission aliquot 10 | 0.386 | 0.283 | 0.10 |
| @SubmissionPerformanceTests @Performance: Submission aliquot 100 | 2.988 | 2.031 | 0.96 |
| @SubmissionPerformanceTests @Performance: Submission aliquot 1000 | 30.541 | 19.594 | 10.95 |
| @SubmissionPerformanceTests @Performance: Submission case 10 | 0.365 | 0.419 | -0.05 |
| @SubmissionPerformanceTests @Performance: Submission case 100 | 3.052 | 2.452 | 0.60 |
| @SubmissionPerformanceTests @Performance: Submission case 1000 | 29.088 | 19.552 | 9.54 |
| @SubmissionPerformanceTests @Performance: Submission clinical_test 10 | 0.694 | 0.359 | 0.34 |
| @SubmissionPerformanceTests @Performance: Submission clinical_test 100 | 3.145 | 2.904 | 0.24 |
| @SubmissionPerformanceTests @Performance: Submission clinical_test 1000 | 31.623 | 22.976 | 8.65 |
| @SubmissionPerformanceTests @Performance: Submission core_metadata_collection 10 | 0.553 | 0.615 | -0.06 |
| @SubmissionPerformanceTests @Performance: Submission core_metadata_collection 100 | 3.265 | 2.263 | 1.00 |
| @SubmissionPerformanceTests @Performance: Submission core_metadata_collection 1000 | 28.849 | 18.952 | 9.90 |
| @SubmissionPerformanceTests @Performance: Submission demographic 10 | 0.415 | 0.358 | 0.06 |
| @SubmissionPerformanceTests @Performance: Submission demographic 100 | 3.278 | 2.301 | 0.98 |
| @SubmissionPerformanceTests @Performance: Submission demographic 1000 | 29.855 | 20.726 | 9.13 |
| @SubmissionPerformanceTests @Performance: Submission diagnosis 10 | 0.46 | 0.346 | 0.11 |
| @SubmissionPerformanceTests @Performance: Submission diagnosis 100 | 3.602 | 2.245 | 1.36 |
| @SubmissionPerformanceTests @Performance: Submission diagnosis 1000 | 33.89 | 24.803 | 9.09 |
| @SubmissionPerformanceTests @Performance: Submission experiment 100 | 2.92 | 2.24 | 0.68 |
| @SubmissionPerformanceTests @Performance: Submission experiment 1000 | 28.838 | 18.465 | 10.37 |
| @SubmissionPerformanceTests @Performance: Submission experimental_metadata 100 | 18.683 | 17.709 | 0.97 |
| @SubmissionPerformanceTests @Performance: Submission exposure 10 | 0.373 | 0.313 | 0.06 |
| @SubmissionPerformanceTests @Performance: Submission exposure 100 | 3.229 | 1.973 | 1.26 |
| @SubmissionPerformanceTests @Performance: Submission exposure 1000 | 29.219 | 21.109 | 8.11 |
| @SubmissionPerformanceTests @Performance: Submission family_history 10 | 0.375 | 0.304 | 0.07 |
| @SubmissionPerformanceTests @Performance: Submission family_history 100 | 3.052 | 1.92 | 1.13 |
| @SubmissionPerformanceTests @Performance: Submission family_history 1000 | 30.16 | 20.677 | 9.48 |
| @SubmissionPerformanceTests @Performance: Submission keyword 10 | 0.576 | 0.287 | 0.29 |
| @SubmissionPerformanceTests @Performance: Submission keyword 100 | 2.859 | 2.006 | 0.85 |
| @SubmissionPerformanceTests @Performance: Submission keyword 1000 | 30.467 | 19.782 | 10.68 |
| @SubmissionPerformanceTests @Performance: Submission publication 10 | 0.36 | 0.32 | 0.04 |
| @SubmissionPerformanceTests @Performance: Submission publication 100 | 2.845 | 2.438 | 0.41 |
| @SubmissionPerformanceTests @Performance: Submission publication 1000 | 28.563 | 19.36 | 9.20 |
| @SubmissionPerformanceTests @Performance: Submission read_group 10 | 0.436 | 0.48 | -0.04 |
| @SubmissionPerformanceTests @Performance: Submission read_group 100 | 2.98 | 2.53 | 0.45 |
| @SubmissionPerformanceTests @Performance: Submission read_group 1000 | 30.922 | 22.273 | 8.65 |
| @SubmissionPerformanceTests @Performance: Submission read_group_qc 10 | 0.557 | 0.446 | 0.11 |
| @SubmissionPerformanceTests @Performance: Submission read_group_qc 100 | 4.552 | 3.366 | 1.19 |
| @SubmissionPerformanceTests @Performance: Submission read_group_qc 1000 | 33.693 | 23.178 | 10.52 |
| @SubmissionPerformanceTests @Performance: Submission sample 10 | 0.464 | 0.372 | 0.09 |
| @SubmissionPerformanceTests @Performance: Submission sample 100 | 3.184 | 2.379 | 0.81 |
| @SubmissionPerformanceTests @Performance: Submission sample 1000 | 31.878 | 22.055 | 9.82 |
| @SubmissionPerformanceTests @Performance: Submission slide 10 | 0.466 | 0.368 | 0.10 |
| @SubmissionPerformanceTests @Performance: Submission slide 100 | 3.026 | 2.284 | 0.74 |
| @SubmissionPerformanceTests @Performance: Submission slide 1000 | 31.385 | 22.098 | 9.29 |
| @SubmissionPerformanceTests @Performance: Submission slide_count 10 | 0.467 | 0.32 | 0.15 |
| @SubmissionPerformanceTests @Performance: Submission slide_count 100 | 2.938 | 2.221 | 0.72 |
| @SubmissionPerformanceTests @Performance: Submission slide_count 1000 | 29.803 | 19.394 | 10.41 |
| @SubmissionPerformanceTests @Performance: Submission slide_image 100 | 17.178 | 18.189 | -1.01 |
| @SubmissionPerformanceTests @Performance: Submission submitted_aligned_reads 100 | 18.364 | 17.096 | 1.27 |
| @SubmissionPerformanceTests @Performance: Submission submitted_copy_number 10 | 1.863 | 1.731 | 0.13 |
| @SubmissionPerformanceTests @Performance: Submission submitted_methylation 10 | 1.899 | 6.99 | -5.09 |
| @SubmissionPerformanceTests @Performance: Submission submitted_unaligned_reads 10 | 2.159 | 1.719 | 0.44 |
| @SubmissionPerformanceTests @Performance: Submission treatment 100 | 2.988 | 2.016 | 0.97 |
| @SubmissionPerformanceTests @Performance: Submission treatment 1000 | 28.526 | 20.421 | 8.10 |

**Comment by PlanXCyborg** [comment]
Jenkins Build 6 : time taken 44 min
Check the https://jenkins.planx-pla.net/job/Performance%20tests%20for%20uc-cdis/job/peregrine/job/PR-138/6/display/redirect


Test results:
	Passed: 109, Failed: 0, Skipped: 0



| Test | Time (`PR`) | Time (`master`) | Diff |
| - | - | - | - |
| @ExportPerformanceTests @Performance: Export all nodes of type: aliquot | 0.274 | 0.111 | 0.16 |
| @ExportPerformanceTests @Performance: Export all nodes of type: case | 0.315 | 0.087 | 0.23 |
| @ExportPerformanceTests @Performance: Export all nodes of type: experiment | 0.28 | 0.314 | -0.03 |
| @ExportPerformanceTests @Performance: Export all nodes of type: program | 0.065 | 0.056 | 0.01 |
| @ExportPerformanceTests @Performance: Export all nodes of type: read_group | 0.462 | 0.132 | 0.33 |
| @ExportPerformanceTests @Performance: Export all nodes of type: read_group_qc | 0.097 | 0.13 | -0.03 |
| @ExportPerformanceTests @Performance: Export all nodes of type: sample | 0.404 | 0.128 | 0.28 |
| @ExportPerformanceTests @Performance: Exporting a record by ID on nodes of type: aliquot | 0.192 | 0.224 | -0.03 |
| @ExportPerformanceTests @Performance: Exporting a record by ID on nodes of type: case | 0.659 | 0.429 | 0.23 |
| @ExportPerformanceTests @Performance: Exporting a record by ID on nodes of type: experiment | 0.384 | 0.327 | 0.06 |
| @ExportPerformanceTests @Performance: Exporting a record by ID on nodes of type: read_group | 0.069 | 0.164 | -0.10 |
| @ExportPerformanceTests @Performance: Exporting a record by ID on nodes of type: read_group_qc | 0.094 | 0.07 | 0.02 |
| @ExportPerformanceTests @Performance: Exporting a record by ID on nodes of type: sample | 0.092 | 0.374 | -0.28 |
| @QueryPerformanceTests @Performance: Executing bottomUp query # 0 | 0.587 | 0.353 | 0.23 |
| @QueryPerformanceTests @Performance: Executing bottomUp query # 1 | 0.804 | 0.268 | 0.54 |
| @QueryPerformanceTests @Performance: Executing bottomUp query # 3 | 0.571 | 0.669 | -0.10 |
| @QueryPerformanceTests @Performance: Executing bottomUp query # 4 | 0.416 | 0.473 | -0.06 |
| @QueryPerformanceTests @Performance: Executing bottomUp query # 5 | 0.469 | 0.305 | 0.16 |
| @QueryPerformanceTests @Performance: Executing bottomUp query # 6 | 0.178 | 0.218 | -0.04 |
| @QueryPerformanceTests @Performance: Executing bottomUp query # 7 | 0.163 | 0.082 | 0.08 |
| @QueryPerformanceTests @Performance: Executing topDown query # 0 | 0.141 | 0.069 | 0.07 |
| @QueryPerformanceTests @Performance: Executing topDown query # 1 | 0.117 | 0.11 | 0.01 |
| @QueryPerformanceTests @Performance: Executing topDown query # 2 | 0.123 | 0.098 | 0.03 |
| @QueryPerformanceTests @Performance: Executing topDown query # 3 | 0.284 | 0.269 | 0.02 |
| @QueryPerformanceTests @Performance: Executing topDown query # 4 | 0.296 | 0.221 | 0.08 |
| @QueryPerformanceTests @Performance: Executing topDown query # 5 | 0.573 | 0.187 | 0.39 |
| @QueryPerformanceTests @Performance: Executing topDown query # 6 | 0.759 | 0.35 | 0.41 |
| @QueryPerformanceTests @Performance: Executing topDown query # 7 | 0.569 | 0.333 | 0.24 |
| @SubmissionPerformanceTests @Performance: Submission acknowledgement 10 | 0.516 | 0.602 | -0.09 |
| @SubmissionPerformanceTests @Performance: Submission acknowledgement 100 | 2.852 | 4.566 | -1.71 |
| @SubmissionPerformanceTests @Performance: Submission acknowledgement 1000 | 29.31 | 40.228 | -10.92 |
| @SubmissionPerformanceTests @Performance: Submission aligned_reads_index 10 | 2.448 | 2.138 | 0.31 |
| @SubmissionPerformanceTests @Performance: Submission aligned_reads_index 100 | 23.925 | 20.195 | 3.73 |
| @SubmissionPerformanceTests @Performance: Submission aliquot 10 | 0.409 | 0.736 | -0.33 |
| @SubmissionPerformanceTests @Performance: Submission aliquot 100 | 3.691 | 4.86 | -1.17 |
| @SubmissionPerformanceTests @Performance: Submission case 10 | 0.395 | 0.78 | -0.38 |
| @SubmissionPerformanceTests @Performance: Submission case 100 | 3.149 | 4.649 | -1.50 |
| @SubmissionPerformanceTests @Performance: Submission case 1000 | 31.32 | 44.871 | -13.55 |
| @SubmissionPerformanceTests @Performance: Submission clinical_test 10 | 0.458 | 0.663 | -0.20 |
| @SubmissionPerformanceTests @Performance: Submission clinical_test 100 | 3.33 | 5.118 | -1.79 |
| @SubmissionPerformanceTests @Performance: Submission clinical_test 1000 | 32.757 | 30.481 | 2.28 |
| @SubmissionPerformanceTests @Performance: Submission core_metadata_collection 10 | 0.65 | 0.693 | -0.04 |
| @SubmissionPerformanceTests @Performance: Submission core_metadata_collection 100 | 3.017 | 4.742 | -1.73 |
| @SubmissionPerformanceTests @Performance: Submission core_metadata_collection 1000 | 30.194 | 44.688 | -14.49 |
| @SubmissionPerformanceTests @Performance: Submission demographic 10 | 0.433 | 0.621 | -0.19 |
| @SubmissionPerformanceTests @Performance: Submission demographic 100 | 3.215 | 5.182 | -1.97 |
| @SubmissionPerformanceTests @Performance: Submission demographic 1000 | 31.058 | 28.838 | 2.22 |
| @SubmissionPerformanceTests @Performance: Submission diagnosis 10 | 0.569 | 0.681 | -0.11 |
| @SubmissionPerformanceTests @Performance: Submission diagnosis 100 | 3.474 | 5.221 | -1.75 |
| @SubmissionPerformanceTests @Performance: Submission experiment 10 | 0.415 | 0.846 | -0.43 |
| @SubmissionPerformanceTests @Performance: Submission experiment 100 | 3.218 | 4.874 | -1.66 |
| @SubmissionPerformanceTests @Performance: Submission experiment 1000 | 30.497 | 42.583 | -12.09 |
| @SubmissionPerformanceTests @Performance: Submission experimental_metadata 10 | 2.508 | 2.174 | 0.33 |
| @SubmissionPerformanceTests @Performance: Submission exposure 10 | 0.546 | 0.642 | -0.10 |
| @SubmissionPerformanceTests @Performance: Submission exposure 100 | 3.525 | 4.765 | -1.24 |
| @SubmissionPerformanceTests @Performance: Submission exposure 1000 | 30.548 | 28.908 | 1.64 |
| @SubmissionPerformanceTests @Performance: Submission family_history 10 | 0.504 | 0.559 | -0.06 |
| @SubmissionPerformanceTests @Performance: Submission family_history 100 | 3.147 | 4.393 | -1.25 |
| @SubmissionPerformanceTests @Performance: Submission family_history 1000 | 30.387 | 28.368 | 2.02 |
| @SubmissionPerformanceTests @Performance: Submission keyword 10 | 0.435 | 0.617 | -0.18 |
| @SubmissionPerformanceTests @Performance: Submission keyword 100 | 3.016 | 4.477 | -1.46 |
| @SubmissionPerformanceTests @Performance: Submission keyword 1000 | 29.199 | 39.666 | -10.47 |
| @SubmissionPerformanceTests @Performance: Submission publication 10 | 0.416 | 0.754 | -0.34 |
| @SubmissionPerformanceTests @Performance: Submission publication 100 | 3.029 | 4.186 | -1.16 |
| @SubmissionPerformanceTests @Performance: Submission publication 1000 | 28.202 | 44.981 | -16.78 |
| @SubmissionPerformanceTests @Performance: Submission read_group 10 | 0.588 | 0.663 | -0.07 |
| @SubmissionPerformanceTests @Performance: Submission read_group 100 | 3.795 | 4.723 | -0.93 |
| @SubmissionPerformanceTests @Performance: Submission read_group_qc 10 | 0.666 | 0.983 | -0.32 |
| @SubmissionPerformanceTests @Performance: Submission read_group_qc 100 | 4.736 | 6.195 | -1.46 |
| @SubmissionPerformanceTests @Performance: Submission sample 10 | 0.406 | 0.8 | -0.39 |
| @SubmissionPerformanceTests @Performance: Submission sample 100 | 3.679 | 4.956 | -1.28 |
| @SubmissionPerformanceTests @Performance: Submission sample 1000 | 34.548 | 32.646 | 1.90 |
| @SubmissionPerformanceTests @Performance: Submission slide 10 | 0.497 | 0.746 | -0.25 |
| @SubmissionPerformanceTests @Performance: Submission slide 100 | 3.117 | 5.261 | -2.14 |
| @SubmissionPerformanceTests @Performance: Submission slide_count 10 | 0.38 | 0.78 | -0.40 |
| @SubmissionPerformanceTests @Performance: Submission slide_count 100 | 3.135 | 4.932 | -1.80 |
| @SubmissionPerformanceTests @Performance: Submission slide_image 100 | 24.232 | 20.112 | 4.12 |
| @SubmissionPerformanceTests @Performance: Submission submitted_aligned_reads 10 | 2.456 | 2.709 | -0.25 |
| @SubmissionPerformanceTests @Performance: Submission submitted_aligned_reads 100 | 24.318 | 19.48 | 4.84 |
| @SubmissionPerformanceTests @Performance: Submission submitted_copy_number 10 | 2.423 | 2.262 | 0.16 |
| @SubmissionPerformanceTests @Performance: Submission submitted_copy_number 100 | 25.522 | 20.069 | 5.45 |
| @SubmissionPerformanceTests @Performance: Submission submitted_methylation 10 | 2.654 | 2.218 | 0.44 |
| @SubmissionPerformanceTests @Performance: Submission submitted_methylation 100 | 23.258 | 20.123 | 3.14 |
| @SubmissionPerformanceTests @Performance: Submission submitted_somatic_mutation 100 | 23.15 | 19.95 | 3.20 |
| @SubmissionPerformanceTests @Performance: Submission submitted_unaligned_reads 10 | 2.383 | 2.187 | 0.20 |
| @SubmissionPerformanceTests @Performance: Submission submitted_unaligned_reads 100 | 24.357 | 19.159 | 5.20 |
| @SubmissionPerformanceTests @Performance: Submission treatment 10 | 0.449 | 0.62 | -0.17 |
| @SubmissionPerformanceTests @Performance: Submission treatment 100 | 3.062 | 4.861 | -1.80 |

File: peregrine/auth/__init__.py [hunk 1]
```diff
@@ -8,56 +8,100 @@
...
-                    flask.current_app.db
-                    .nodes(models.Program)
-                    .props(name=program)
-                    .scalar()
+logger = get_logger(__name__)
+
+
+def resource_path_to_project_ids(resource_path):
+    parts = resource_path.strip('/').split('/')
+    if resource_path == "/" or (parts and parts[0] == "programs"):
```
[the remark this answers is not in the document]
Comment (paulineribeyre) [reply peregrine/auth/__init__.py]: yes thank you!!
except it's not a bad resource path, it's just ignored by peregrine (eg /data_upload or /consent/xxx)

**Comment by PlanXCyborg** [comment]
Jenkins Build 8 : time taken 37 min
Check the https://jenkins.planx-pla.net/job/Performance%20tests%20for%20uc-cdis/job/peregrine/job/PR-138/8/display/redirect


Test results:
	Passed: 109, Failed: 0, Skipped: 0



| Test | Time (`PR`) | Time (`master`) | Diff |
| - | - | - | - |
| @ExportPerformanceTests @Performance: Export all nodes of type: aliquot | 0.284 | 0.248 | 0.04 |
| @ExportPerformanceTests @Performance: Export all nodes of type: program | 0.113 | 0.052 | 0.06 |
| @ExportPerformanceTests @Performance: Export all nodes of type: project | 0.094 | 0.043 | 0.05 |
| @ExportPerformanceTests @Performance: Export all nodes of type: read_group | 0.405 | 0.476 | -0.07 |
| @ExportPerformanceTests @Performance: Export all nodes of type: read_group_qc | 0.095 | 0.11 | -0.02 |
| @ExportPerformanceTests @Performance: Export all nodes of type: sample | 0.414 | 0.655 | -0.24 |
| @ExportPerformanceTests @Performance: Exporting a record by ID on nodes of type: aliquot | 0.169 | 0.254 | -0.09 |
| @ExportPerformanceTests @Performance: Exporting a record by ID on nodes of type: case | 0.283 | 0.264 | 0.02 |
| @ExportPerformanceTests @Performance: Exporting a record by ID on nodes of type: experiment | 0.324 | 0.192 | 0.13 |
| @ExportPerformanceTests @Performance: Exporting a record by ID on nodes of type: read_group | 0.06 | 0.118 | -0.06 |
| @ExportPerformanceTests @Performance: Exporting a record by ID on nodes of type: read_group_qc | 0.084 | 0.078 | 0.01 |
| @ExportPerformanceTests @Performance: Exporting a record by ID on nodes of type: sample | 0.144 | 0.093 | 0.05 |
| @QueryPerformanceTests @Performance: Executing bottomUp query # 1 | 0.636 | 0.728 | -0.09 |
| @QueryPerformanceTests @Performance: Executing bottomUp query # 2 | 0.661 | 0.798 | -0.14 |
| @QueryPerformanceTests @Performance: Executing bottomUp query # 3 | 0.513 | 0.65 | -0.14 |
| @QueryPerformanceTests @Performance: Executing bottomUp query # 4 | 0.42 | 0.382 | 0.04 |
| @QueryPerformanceTests @Performance: Executing bottomUp query # 5 | 0.416 | 0.44 | -0.02 |
| @QueryPerformanceTests @Performance: Executing bottomUp query # 7 | 0.112 | 0.077 | 0.04 |
| @QueryPerformanceTests @Performance: Executing topDown query # 0 | 0.13 | 0.112 | 0.02 |
| @QueryPerformanceTests @Performance: Executing topDown query # 1 | 0.124 | 0.086 | 0.04 |
| @QueryPerformanceTests @Performance: Executing topDown query # 2 | 0.159 | 0.185 | -0.03 |
| @QueryPerformanceTests @Performance: Executing topDown query # 3 | 0.248 | 0.233 | 0.02 |
| @QueryPerformanceTests @Performance: Executing topDown query # 4 | 0.423 | 0.397 | 0.03 |
| @QueryPerformanceTests @Performance: Executing topDown query # 5 | 0.391 | 0.443 | -0.05 |
| @QueryPerformanceTests @Performance: Executing topDown query # 6 | 0.64 | 0.903 | -0.26 |
| @QueryPerformanceTests @Performance: Executing topDown query # 7 | 0.862 | 0.706 | 0.16 |
| @SubmissionPerformanceTests @Performance: Submission acknowledgement 10 | 0.391 | 0.437 | -0.05 |
| @SubmissionPerformanceTests @Performance: Submission aligned_reads_index 10 | 2.16 | 1.72 | 0.44 |
| @SubmissionPerformanceTests @Performance: Submission aligned_reads_index 100 | 22.722 | 15.996 | 6.73 |
| @SubmissionPerformanceTests @Performance: Submission aliquot 10 | 0.527 | 0.426 | 0.10 |
| @SubmissionPerformanceTests @Performance: Submission case 10 | 0.485 | 0.417 | 0.07 |
| @SubmissionPerformanceTests @Performance: Submission clinical_test 1000 | 32.849 | 34.607 | -1.76 |
| @SubmissionPerformanceTests @Performance: Submission core_metadata_collection 100 | 3.114 | 3.655 | -0.54 |
| @SubmissionPerformanceTests @Performance: Submission diagnosis 10 | 0.508 | 0.559 | -0.05 |
| @SubmissionPerformanceTests @Performance: Submission experiment 10 | 0.543 | 0.672 | -0.13 |
| @SubmissionPerformanceTests @Performance: Submission experiment 100 | 3.653 | 3.04 | 0.61 |
| @SubmissionPerformanceTests @Performance: Submission experimental_metadata 10 | 2.189 | 1.738 | 0.45 |
| @SubmissionPerformanceTests @Performance: Submission experimental_metadata 100 | 20.684 | 16.371 | 4.31 |
| @SubmissionPerformanceTests @Performance: Submission exposure 100 | 2.99 | 3.591 | -0.60 |
| @SubmissionPerformanceTests @Performance: Submission exposure 1000 | 30.072 | 33.067 | -3.00 |
| @SubmissionPerformanceTests @Performance: Submission family_history 10 | 0.453 | 0.534 | -0.08 |
| @SubmissionPerformanceTests @Performance: Submission keyword 10 | 0.394 | 0.462 | -0.07 |
| @SubmissionPerformanceTests @Performance: Submission keyword 100 | 3.929 | 3.066 | 0.86 |
| @SubmissionPerformanceTests @Performance: Submission keyword 1000 | 28.418 | 30.235 | -1.82 |
| @SubmissionPerformanceTests @Performance: Submission publication 100 | 3.927 | 3.199 | 0.73 |
| @SubmissionPerformanceTests @Performance: Submission publication 1000 | 28.495 | 30.266 | -1.77 |
| @SubmissionPerformanceTests @Performance: Submission read_group 10 | 0.559 | 0.503 | 0.06 |
| @SubmissionPerformanceTests @Performance: Submission read_group 100 | 4.063 | 3.636 | 0.43 |
| @SubmissionPerformanceTests @Performance: Submission read_group_qc 10 | 0.867 | 0.657 | 0.21 |
| @SubmissionPerformanceTests @Performance: Submission read_group_qc 100 | 5.105 | 5.623 | -0.52 |
| @SubmissionPerformanceTests @Performance: Submission read_group_qc 1000 | 34.268 | 38.55 | -4.28 |
| @SubmissionPerformanceTests @Performance: Submission sample 10 | 0.496 | 0.634 | -0.14 |
| @SubmissionPerformanceTests @Performance: Submission sample 100 | 3.398 | 3.128 | 0.27 |
| @SubmissionPerformanceTests @Performance: Submission slide_count 10 | 0.508 | 0.453 | 0.06 |
| @SubmissionPerformanceTests @Performance: Submission slide_count 1000 | 29.953 | 31.535 | -1.58 |
| @SubmissionPerformanceTests @Performance: Submission slide_image 10 | 2.382 | 1.827 | 0.55 |
| @SubmissionPerformanceTests @Performance: Submission slide_image 100 | 21.967 | 16.661 | 5.31 |
| @SubmissionPerformanceTests @Performance: Submission submitted_aligned_reads 10 | 2.156 | 1.812 | 0.34 |
| @SubmissionPerformanceTests @Performance: Submission submitted_aligned_reads 100 | 21.087 | 17.255 | 3.83 |
| @SubmissionPerformanceTests @Performance: Submission submitted_copy_number 10 | 2.223 | 1.632 | 0.59 |
| @SubmissionPerformanceTests @Performance: Submission submitted_copy_number 100 | 22.128 | 15.807 | 6.32 |
| @SubmissionPerformanceTests @Performance: Submission submitted_methylation 10 | 2.259 | 1.875 | 0.38 |
| @SubmissionPerformanceTests @Performance: Submission submitted_methylation 100 | 21.239 | 15.974 | 5.27 |
| @SubmissionPerformanceTests @Performance: Submission submitted_somatic_mutation 10 | 2.221 | 1.701 | 0.52 |
| @SubmissionPerformanceTests @Performance: Submission submitted_somatic_mutation 100 | 20.764 | 15.416 | 5.35 |
| @SubmissionPerformanceTests @Performance: Submission submitted_unaligned_reads 10 | 2.347 | 1.77 | 0.58 |
| @SubmissionPerformanceTests @Performance: Submission submitted_unaligned_reads 100 | 20.174 | 16.467 | 3.71 |
| @SubmissionPerformanceTests @Performance: Submission treatment 1000 | 30.251 | 32.256 | -2.01 |

**Pull request merged**